### PR TITLE
AWS S3 file upload endpoint

### PR DESCRIPTION
**Pull request by mvoloder**
Create file upload endpoint on StarApi

[Task link](http://the-shop.io:3000/projects/586016083e5bbe768349a4b0/sprints/58601d6c3e5bbe4cb74ea61f/tasks/5863dbd53e5bbe318101d43a)

## Checklist

- [ ] README.md updated
- [x] Tests covered

## Environment updates
added S3_KEY, S3_SECRET, S3_BUCKET, S3_REGION

"league/flysystem-aws-s3-v3": "^1.0" --->should be in composer.json

## Test notes

you sent screenshot

## Screenshots

If applicable, add screenshots of what you did.

### CODE REVIEW

File: app/Http/routes.php [hunk 1]
```diff
@@ -51,6 +49,7 @@
         Route::put('trello/board/{boardId}/ticket/{ticketId}/member/{memberId}/remove', 'TrelloController@removeMember');
         Route::put('trello/board/{boardid}/ticket/{id}', 'TrelloController@setDueDate');
         Route::get('configuration', 'ConfigurationController@getConfiguration');
+        Route::get('upload', 'FileUploadController@uploadFile');
```
Comment (lotar) [inline app/Http/routes.php]: should be `::post(` - we're creating a new record

File: app/Http/Controllers/FileUploadController.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public function uploadFile(Request $request)
+    {
+        $file = $request->file('image');
+
+        $file_name = time() . "." . $file->getClientOriginalExtension();
+
+        $s3 = Storage::disk('s3');
+        $file_path = '/starapi/tests/' . $file_name;
+        $s3->put($file_path, file_get_contents($file), 'public');
```
Comment (lotar) [inline app/Http/Controllers/FileUploadController.php]: We need to store this in own collection `uploads` - how else would we be able to fetch the data from S3?

Each document should have `userId` and `fileUrl` attributes attached to it.

**Review by lotar — CHANGES_REQUESTED** [review]

**Comment by lotar** [comment]
Also, please merge in from master / resolve conflicts so that it can be merged.

File: app/Http/Controllers/GenericResourceController.php [hunk 3]
```diff
@@ -145,6 +147,13 @@ public function update(Request $request)
             event(new TaskUpdateSlackNotify($tasks));
         }
 
+
+        $a = $model;
```
Comment (lotar) [inline app/Http/Controllers/GenericResourceController.php]: What's `$a`?

File: app/Http/Controllers/FileUploadController.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        $files = $request->file();
+
+        foreach ($files as $file) {
+            GenericModel::setCollection('uploads');
+            $upload = GenericModel::create();
+
+            $file_name = time() . "." . $file->getClientOriginalExtension();
+
+            $s3 = Storage::disk('s3');
+            $file_path = '/starapi/tests/' . $file_name;
```
Comment (lotar) [inline app/Http/Controllers/FileUploadController.php]: remove '/starapi/tests'

File: app/Http/Controllers/FileUploadController.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            $project_id = null;
+        }
+
+        $files = $request->file();
+
+        foreach ($files as $file) {
+            GenericModel::setCollection('uploads');
+            $upload = GenericModel::create();
+
+            $file_name = time() . "." . $file->getClientOriginalExtension();
```
Comment (lotar) [inline app/Http/Controllers/FileUploadController.php]: Camel case variables

File: app/Events/TaskUpdate.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Get the channels the event should be broadcast on.
+     *
+     * @return array
+     */
+    public function broadcastOn()
+    {
+        return [];
+    }
+}
```
Comment (lotar) [inline app/Events/TaskUpdate.php]: newline missing

File: app/Http/Controllers/FileUploadController.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            $fileUrl = Storage::url($file);
+
+            $upload->profileId = $user_id;
+            $upload->projectId = $project_id;
+            $upload->name = $file_name;
+            $upload->fileUrl = $fileUrl;
+            $upload->save();
+        }
+    }
+}
```
Comment (lotar) [inline app/Http/Controllers/FileUploadController.php]: newline missing

File: app/Http/Controllers/FileUploadController.php [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            GenericModel::setCollection('uploads');
+            $upload = GenericModel::create();
+
+            $file_name = time() . "." . $file->getClientOriginalExtension();
+
+            $s3 = Storage::disk('s3');
+            $file_path = '/starapi/tests/' . $file_name;
+            $s3->put($file_path, file_get_contents($file), 'public');
+
+            $fileUrl = Storage::url($file);
```
Comment (lotar) [inline app/Http/Controllers/FileUploadController.php]: does `$fileUrl` point to S3 url or?

**Review by lotar — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/FileUploadController.php (hunk 8 above)
Comment (mvoloder) [inline app/Http/Controllers/FileUploadController.php]: local storage, fixed to point on to S3

**Review by lotar — APPROVED** [review]

**Pull request merged**